### DISCUSSION TITLE
Geometrically necessary dislocation

### DISCUSSION THREAD

**Discussion by mugikumar** [mtex-toolbox/mtex#1728] [Ask Anything] [answered]
Hi all
   Thanks for giving the mtex code to evaluate the ebsd data. i have been using the GND code to plot it. i have to know the code of average gnd value. 
![0 0 100 100](https://user-images.githubusercontent.com/109027461/215464704-751bec09-5064-4493-b846-ad130590291a.png)

**Comment by kilir** [comment, marked as answer]
Hi,
you might have to specify what you want to do. Average over what? Grain, entire area,...?

Cheers,
Rüdiger

**Reply by mugikumar** [reply]
Hi kilir,
        Thanks for your reply, I have to know the average geometrically necessary dislocation density of the entire area.

**Reply by kilir** [reply]
Assuming what you did was that you followed the documentation, you can compute an average from `factor*sum(abs(rho .* dSRot.u),2)`, e.g. using `mean()`.
Cheers,
Rüdiger

**Reply by mugikumar** [reply]
hi Kilir,
           Thanks for suggesting the code, but i  tried this code before itself it was showing  mean((factor*sum(abs(rho .* dsRot.u),2)))

ans =

   NaN
Could you please help me out
with regards
Murugesh kumar

**Reply by kilir** [reply]
In the case you want the arithmetic mean but you have Nan in your input, try `nanmean()`.

**Reply by mugikumar** [reply]
Thanks, kilir yes it's working.

**Reply by alone19990909** [reply]
Hi

**Reply by alone19990909** [reply]
hi , thank you so much guys. i had the same problem with that and your comment was so helpful
